### PR TITLE
luhn: add a test case

### PR DESCRIPTION
**Pull request by idealhack**
https://github.com/exercism/problem-specifications/blob/master/exercises/luhn/description.md

### CODE REVIEW

File: exercises/luhn/canonical-data.json [hunk 1]
```diff
@@ -105,6 +105,14 @@
         "value": "091"
       },
       "expected": true
+    },
+    {
+      "description": "strings with non-digits is invalid",
```
Comment (rpottsoh) [inline exercises/luhn/canonical-data.json]: How is this new case different from `"valid strings with a non-digit included become invalid"`?

Comment (idealhack) [reply exercises/luhn/canonical-data.json]:
```go
for i, char := range str {
    n := int(char - '0')
    if (len(str)-i)%2 == 0 {
        n *= 2
        if n >= 10 {
            n -= 9
        }
    }
    sum += n
}
```

The code above will pass current tests, but not this one.

So we have to check non-digits by adding:

```go
if n > 9 || n < 0 {
    return false
}
```

**Comment by idealhack** [comment]
@rpottsoh I've increased the minor version number, please take a look

File: exercises/luhn/canonical-data.json (hunk 1 above)
Comment (rpottsoh) [reply exercises/luhn/canonical-data.json]: Thanks for the explanation.  I can see now how `"valid strings with a non-digit included become invalid"` passes without the new test case.

Other reviewers may ask that your new test case be moved to an earlier location in the JSON file.  I'll leave that open to others to address.

@idealhack thanks for your time in improving this exercise! :+1:

**Comment by rpottsoh** [comment]
`version` looks good.  I am going to let this PR sit for now, to allow time for others to weigh in the change you are proposing.

**Comment by idealhack** [comment]
@rpottsoh You are welcome. Exercism is great because of your work.

**Pull request merged**